### DISCUSSION TITLE
Can I use prisma to subscribe to PostgreSQL changes?

### DISCUSSION THREAD

**Discussion by xzilja** [prisma/prisma#2923] [Q&A] [answered]
Hey everyone 👋 

I'm currently building an endpoint to send server side events (SSE) to the client and what I have to do right now is something like

1. User makes a post request
2. prisma client updates database and reads new updated data
3. new data is emitted with nodejs event emitter
4. SSE endpoint listens for these events and then pushes new data to the client

This has some overhead and drawbacks i.e. there are no transactions in prisma atm, so step 2 executes separate operations and  of course I need to manage event emiters.

A much better alternative to this would be to use prisma to subscribe to changes in a specific row inside that SSE endpoint and push new data when  it is received.

I looked at the docs, but wasn't able to find anything like this. So I guess my question is if its possible or maybe this is a good feature proposal?

**Comment by ryands17** [comment, marked as answer]
Hey @IljaDaderko :wave: 
Yes indeed this is a good feature proposal and you can find it [here](https://github.com/prisma/prisma/issues/905). There is a main issue linked inside which provides the entire basis for Subscriptions and how they can be implemented. Do add a :+1: on the issue as well so that we can judge the priority of those issues.

The workaround for now would be as you mentioned to use an Event Emitter or PubSub service at the application layer to handle subscriptions to the changes in the database.

**Comment by flyingkaktus** [comment]
So 4 years later, how is it going? still no subscribe to postgres events?

**Reply by dataf3l** [reply]
hi i need pubsub potgres prisma thing how can i help

**Comment by ereinach** [comment]
+1

**Comment by rgorai-cc** [comment]
Bump

**Comment by wallwhite** [comment]
+1

**Comment by Saliheen** [comment]
bump

**Comment by DongMadison** [comment]
I was here

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
